### PR TITLE
Preliminary Linux-OpenBSD cross-compile support.

### PR DESCRIPTION
**Pull request by 3405691582**
With a properly prepared sysroot and toolchain file, these changes permit
cross-compilation of Swift as well as LLVM, CMark, and Dispatch (picking,
as usual, apple/swift-corelibs-libdispatch#556) from a Linux host
generating OpenBSD binaries.

The toolchain file must be specified as an environment variable to
`build-script` and discussion on how to properly set up the sysroot and
toolchain file will be handled later.

### CODE REVIEW

**Comment by kateinoigakukun** [comment]
@swift-ci Please smoke test

File: cmake/modules/SwiftConfigureSDK.cmake [hunk 1]
```diff
@@ -404,6 +404,11 @@ macro(configure_sdk_unix name architectures)
         message(STATUS "OpenBSD Version: ${openbsd_system_version}")
 
         set(SWIFT_SDK_OPENBSD_ARCH_amd64_TRIPLE "amd64-unknown-openbsd${openbsd_system_version}")
+
+	if(NOT CMAKE_SYSROOT STREQUAL "")
```
Comment (compnerd) [inline cmake/modules/SwiftConfigureSDK.cmake:408]:
```suggestion
	if(CMAKE_SYSROOT)
```

File: utils/build-script-impl [hunk 2]
```diff
@@ -1887,6 +1894,13 @@ for host in "${ALL_HOSTS[@]}"; do
                     )
                 fi
 
+                if [[ $(is_cross_tools_host ${host}) && "${host}" == "openbsd-"* && "${OPENBSD_USE_TOOLCHAIN_FILE}" != "" ]]; then
```
Comment (compnerd) [inline utils/build-script-impl:1897]:
```suggestion
                if [[ $(is_cross_tools_host ${host}) && "${host}" == "openbsd-"* && -n "${OPENBSD_USE_TOOLCHAIN_FILE}" ]]; then
```

File: utils/build-script-impl [hunk 3]
```diff
@@ -1605,6 +1605,13 @@ for host in "${ALL_HOSTS[@]}"; do
                 -DCMAKE_BUILD_WITH_INSTALL_RPATH="1"
             )
         fi
+
+        if [[ "${host}" == "openbsd-"* && "${OPENBSD_USE_TOOLCHAIN_FILE}" != "" ]]; then
```
Comment (compnerd) [inline utils/build-script-impl:1609]:
```suggestion
        if [[ "${host}" == "openbsd-"* && -n "${OPENBSD_USE_TOOLCHAIN_FILE}" ]]; then
```

**Review by compnerd — APPROVED** [review]

**Comment by compnerd** [comment]
CC: @etcwilde

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Pull request merged**